2025-Code/src/main/java/frc/robot/subsystems/Limelight.java

```diff
@@ -199,21 +199,23 @@ private void updateOdometryLL2() {
     m_lastFrame2 = frame2;
   }
 
+  private boolean hasReefTag() {
+    return LimelightHelpers.getRawFiducials(LIMELIGHT_1).length > 0;
+  }
+
+  private boolean hasPlayerStationTag() {
+    return LimelightHelpers.getRawFiducials(LIMELIGHT_2).length > 0;
+  }
+
   private void updateOdometry() {
+    boolean hasReefTag = hasReefTag();
+    boolean hasPlayerStationTag = hasPlayerStationTag();
+
     m_swerve.setVisionMeasurementStdDevs(VecBuilder.fill(0.7, 0.7, 9999999));
-    int tagCt1 = LimelightHelpers.getRawFiducials(LIMELIGHT_1).length;
-    int tagCt2 = LimelightHelpers.getRawFiducials(LIMELIGHT_2).length;
-    if (tagCt1 + tagCt2 < 2) {
-      if (tagCt1 > 0) {
-        updateOdometryLL1();
-      } else {
-        updateOdometryLL2();
-      }
-      return;
-    }
-    if (Subsystem.coral.hasCoral()) {
+
+    if ((Subsystem.coral.hasCoral() && hasPlayerStationTag) || (hasReefTag && !hasPlayerStationTag)) {
```

Comment:
should this be (Subsystem.coral.hasCoral() && hasReefTag) ??

Reply:
perhaps:
`    if (hasReefTag && (Subsystem.coral.hasCoral() || !hasPlayerStationTag))`